### PR TITLE
[v8] Replace read keychain API with completion handler based API

### PR DESCRIPTION
**Pull request by ncooke3**
I'm merging into a feature branch off of the v8 branch. I'm breaking the work into smaller PRs to make each API re-write easier to review.

### CODE REVIEW

**Review by paulb777 — APPROVED** [review]
Swift would be so much better for this! 

LGTM, thanks!

**Review by andrewheard — APPROVED** [review]

File: GoogleUtilities/Environment/SecureStorage/GULKeychainStorage.m [hunk 1]
```diff
@@ -126,40 +124,38 @@ - (instancetype)initWithService:(NSString *)service cache:(NSCache *)cache {
...
+                        accessGroup:(nullable NSString *)accessGroup
+                  completionHandler:(void (^)(id<NSSecureCoding> _Nullable obj, NSError * _Nullable error))completionHandler {
+    // Look for the object in the keychain.
+    dispatch_async(self.keychainQueue, ^{
+        NSDictionary *query = [self keychainQueryWithKey:key accessGroup:accessGroup];
+        NSError *error;
+        NSData *encodedObject = [GULKeychainUtils getItemWithQuery:query error:&error];
+
+        if (error) {
+          completionHandler(nil, error);
```
Comment (ncooke3) [inline GoogleUtilities/Environment/SecureStorage/GULKeychainStorage.m:138]: This line needed to be followed with a `return`.

File: GoogleUtilities/Environment/SecureStorage/GULKeychainStorage.m [hunk 2]
```diff
@@ -126,40 +124,38 @@ - (instancetype)initWithService:(NSString *)service cache:(NSCache *)cache {
...
+        }
+        if (!encodedObject) {
+          completionHandler(nil, nil);
```
Comment (ncooke3) [inline GoogleUtilities/Environment/SecureStorage/GULKeychainStorage.m:141]: This line needed to be followed with a `return`.

File: GoogleUtilities/Environment/SecureStorage/GULKeychainStorage.m [hunk 3]
```diff
@@ -126,40 +124,38 @@ - (instancetype)initWithService:(NSString *)service cache:(NSCache *)cache {
...
+        }
+        id object = [NSKeyedUnarchiver unarchivedObjectOfClass:objectClass
+                                                      fromData:encodedObject
+                                                         error:&error];
+        if (error) {
+          completionHandler(nil, error);
```
Comment (ncooke3) [inline GoogleUtilities/Environment/SecureStorage/GULKeychainStorage.m:147]: This line needed to be followed with a `return`.

**Review by ncooke3 — COMMENTED** [review]
Caught this while working on the next PR. Completion handlers never cease to be tricky!

**Pull request merged**